src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -136,6 +147,65 @@ def _store_area_penalties(self, results_dict, area):
         results_dict["totals_with_penalties"] = \
             results_dict["total_cost"] + results_dict["penalty_cost"]
 
+    @property
+    def cumulative_bills(self):
+        return {
+            uuid: {
+                "name": results["name"],
+                "spent_total": round_floats_for_ui(results['spent_total']),
+                "earned": round_floats_for_ui(results['earned']),
+                "penalties": round_floats_for_ui(results['penalties']),
+                "total": round_floats_for_ui(results['total'])
+            }
+            for uuid, results in self.cumulative_bills_results.items()
+        }
+
+    def update_cumulative_bills(self, area):
+        for child in area.children:
+            self.update_cumulative_bills(child)
+
+        if area.uuid not in self.cumulative_bills_results or \
+                ConstSettings.GeneralSettings.KEEP_PAST_MARKETS is True:
+            self.cumulative_bills_results[area.uuid] = {
+                "name": area.name,
+                "spent_total": 0.0,
+                "earned": 0.0,
+                "penalties": 0.0,
+                "total": 0.0,
+            }
+
+        if area.strategy is None:
+            all_child_results = [self.cumulative_bills_results[c.uuid]
+                                 for c in area.children]
+            self.cumulative_bills_results[area.uuid] = {
+                "name": area.name,
+                "spent_total": sum(c["spent_total"] for c in all_child_results),
+                "earned": sum(c["earned"] for c in all_child_results),
+                "penalties": sum(c["penalties"] for c in all_child_results
+                                 if c["penalties"] is not None),
+                "total": sum(c["total"] for c in all_child_results),
+            }
+        else:
+            trades = [m.trades
+                      for m in self._get_past_markets_from_area(area.parent, "past_markets")]
+            trades = list(chain(*trades))
+
+            spent_total = sum(trade.offer.price
+                              for trade in trades
+                              if trade.buyer == area.name) / 100.0
+            earned = sum(trade.offer.price - trade.fee_price
+                         for trade in trades
+                         if trade.seller == area.name) / 100.0
+            penalty_energy = self._calculate_device_penalties(area)
+            if penalty_energy is None:
+                penalty_energy = 0.0
+            penalties = penalty_energy * DEVICE_PENALTY_RATE / 100.0
```

Comment:
please rename to `penalty_cost` or similar. Thanks!

Reply:
Fixing it thanks